ts/metrics/metric_cache.py

```diff
@@ -0,0 +1,209 @@
+"""
+1. Parse model_metrics portion of yaml file and get the data
+2. Create Metric objects based off of the model_metrics yaml data
+3. Create hash table with key being metricType_metricName_dimensions
+and value being the respective Metric object created in the previous step
+
+naming and unit testing
+"""
+import sys
+import yaml
+import psutil
+
+from ts.metrics.metric import Metric
+from ts.metrics.dimension import Dimension
+
+
+class MetricsCache:
+    def __init__(self, yaml_file):
+        """
+        Constructor for MetricsCaching class
+
+            MetricsCaching class will hold Metric objects in a cache that can be edited and read from.
+            The class also has the ability to parse a yaml file into Metrics objects
+
+        Parameters
+        ----------
+        yaml_file: str
```

Comment:
I'm wondering if we should create a more abstract metric cache and then have as one of the concrete implementations one for YAML. It'll let us future proof in case we decide another format is better in the future

Reply:
Yeah that makes sense to me, I'll give it a go